widget/contrib/updates.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+
+     Licensed under GNU General Public License v2
+      * (c) 2017, Alphonse Mariyagnanaseelan
+
+--]]
+
+local helpers       = require("lain.helpers")
+local awful         = require("awful")
+local shell         = require("awful.util").shell
+local wibox         = require("wibox")
+local naughty       = require("naughty")
+local lines, floor  = io.lines, math.floor
+local string        = { format = string.format,
+                        gsub   = string.gsub,
+                        len    = string.len}
+
+-- Available updates
+-- lain.widget.contrib.updates
+
+local function factory(args)
+    local updates          = { widget = wibox.widget.textbox() }
+    local args            = args or {}
+    local timeout         = args.timeout or 900
+
+    local notify          = args.notify or false
+    local notify_title    = args.notify_title or "Updates"
+    local notify_timeout  = args.notify_timeout or 15
+
+    local settings        = args.settings or function() end
+    local package_manager = args.package_manager
+    local command         = args.command or ""
+
+    local commands = {
+        pacman        = "checkupdates | sed 's/->/→/' | column -t",
+        pacaur        = "pacaur -k --color never | sed 's/:: [a-zA-Z0-9]\\+ //' | sed 's/->/→/' | column -t",
+        pacman_pacaur = "( checkupdates & pacaur -k --color never | sed 's/:: [a-zA-Z0-9]\\+ //' ) | sed 's/->/→/' | sort | column -t",
+        dnf           = "dnf check-update --quiet",
+        apt           = "apt-show-versions -u"
+    }
```

Comment:
I am not sure about this command table either. Potentially, since it should be as comprehensive as possible, it could grow in size without limits. Better would be to specify the command as argument, and to provide some examples in the wiki entry.

Reply:
That was something I considered for convenience. But you're right, other than that potential users will most likely figure this out.